### DISCUSSION TITLE
Free and open-source payments orchestrator/system, Hyperswitch

### DISCUSSION THREAD

**Discussion by NamelessGO** [WeblateOrg/weblate#11091] [Ideas] [answered]
Weblate uses a proprietary bi-language website: https://www.thepay.eu/, found in https://weblate.org/donate/

[Hyperswitch](https://hyperswitch.io/) is built with **Rust** and is now fully free and open source (app server, card vault, web client SDK & dashboard). It is optimized for sub 30ms overhead with 50+ processor integrations. There's also multiple self-hosting options

[Reddit Questions and Answers with one of the devs
](https://www.reddit.com/r/selfhosted/comments/18rzd4a/hyperswitch_free_and_opensource_payments/
)

> As of Sept 2023, we support 50+ payment processors and multiple global payment methods. In addition, we are continuously integrating new processors based on their reach and community requests. Our target is to support 100+ processors by H2 2023. You can find the latest list of payment processors, supported methods, and features [here](https://hyperswitch.io/pm-list).

[Source Code](https://github.com/juspay/hyperswitch): 9.2k stars, 730 forks, 158 contributors

![image](https://github.com/WeblateOrg/weblate/assets/66227691/3553e58c-ae4f-4d64-a8d0-30843b005125)

**Comment by nijel** [comment, marked as answer]
AFAIK, any credit card processing will be done by a proprietary website. Having an open-source middleman doesn't really change much. Furthermore, having a middleman without an official Python bindings is not really helpful for us.

Any abstraction layer for payments I've seen so far sucks in recurring payments. The reason is simple - each processor handles this very differently, and building abstraction here is tricky. https://docs.hyperswitch.io/features/payment-flows-and-management/mandates-and-recurring-payments really doesn't explain this much into detail.

PS: We will most likely switch to some other processor this year for various reasons.
